File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int Max=0;
        int l=0;
        int n=height.size();
        int h=n-1;
        while(l<=h){
            int boundaryHeight=min(height[l],height[h]);
            Max=max(Max, boundaryHeight*(h-l));
            if(height[l]<height[h]){
                l++;
            }else{h--;}
        }
        return Max;
    }
};

File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        Max=0
        left,right=0,len(height)-1
        while(left<right):
            if height[left]<height[right]:
                Max=max(Max,(right-left)*height[left])
                left+=1
            else:
                Max=max(Max,(right-left)*height[right])
                right-=1
        return Max